### ISSUE TITLE
nginx reverse proxy with multiple nodes and auth_basic

### ISSUE THREAD

**Issue by TBNolan** [netdata/netdata#8947]
Is it possible to run multiple netdata instances on different machines, all set up with nginx reverse proxy using `auth_basic`, and manage them from the node console?

I have successfully set up 3 machines, all with different FQDNs and using `auth_basic`. However, they can't talk to each other from `console.html` because they unsurprisingly can't access each others' API behind the `auth_basic`.

Does anyone know a workaround or solution to this? Do I have to forfeit the node console if I want to have `auth_basic` on each install?

**Comment by TBNolan** [comment]
Here is a sample nginx site config
```
upstream backend {
        # Netdata server
        server 127.0.0.1:19999;
        keepalive 64;
}

server {

        server_name netdata1.mydomain.com;

        #auth_basic "Protected";
        #auth_basic_user_file passwords;

        location / {
                proxy_set_header X-Forwarded-Host $host;
                proxy_set_header X-Forwarded-Server $host;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_pass http://backend;
                proxy_http_version 1.1;
                proxy_pass_request_headers on;
                proxy_set_header Connection "keep-alive";
                proxy_store off;
        }

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/netdata1.mydomain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/netdata1.mydomain.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}



server {
    if ($host = netdata1.mydomain.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot



        server_name netdata1.mydomain.com;
    listen 80;
    return 404; # managed by Certbot


}
```

**Comment by thiagoftsm** [comment]
Hi @TBNolan ,

> Does anyone know a workaround or solution to this? Do I have to forfeit the node console if I want to have `auth_basic` on each install?

Netdata is working now to bring a new cloud environment and we hope this will fix these kinds of problems, but for you do not wait it to be released, I was thinking that a possible solution for this would be you centralize all Netdata in local registry inside your network, did you already try this?

Best regards!

**Comment by TBNolan** [comment]
Thanks for your response @thiagoftsm 

That's an interesting idea and I'll have to look more into it. Is this the implementation you're referring to? [Netdata: Run your own registry](https://github.com/netdata/netdata/tree/master/registry#run-your-own-registry)

Looking at it briefly, it looks like I'll need each of my nodes to announce to a chosen one of my servers. Since they are all still protected by `auth_basic` I'd need to use their non-proxied announce url, i.e. `http://my-server-fqdn.com:19999`. I currently have that port firewalled off to outside traffic. Will I need to open it up to my other servers or is there a better way?

I'll need to dig in more, but if you have any resources for where to start, I'd be grateful.

Thank you!

**Comment by thiagoftsm** [comment]
> That's an interesting idea and I'll have to look more into it. Is this the implementation you're referring to?

Yes, this is my suggestion.

> Looking at it briefly, it looks like I'll need each of my nodes to announce to a chosen one of my servers.

This would be the first step.

> I currently have that port firewalled off to outside traffic. 

When I gave the suggestion, I imagined a scenario where all hosts with Nedata were behind an unique firewall, but they could communicate direct among them.

> Will I need to open it up to my other servers or is there a better way?

I will give a suggestion for you, please, verify whether this could be a good road for you. Netdata allows us to open different ports and specify what we can do in that port, for example, in my host I use the following bind to:

```
bind to = *=dashboard|registry|streaming|management|netdata.conf|badges *:20000=dashboard|registry|streaming|netdata.conf|badges|management^SSL=optional
```

we could change this for something like this:

```
bind to = *=dashboard|registry|streaming|management|netdata.conf|badges *:20000=registry
```

With this last configuration, we would have Netdata running with the port 20000 open only to receive `registry` requests. 

you are welcome!

best regards!

**Comment by TBNolan** [comment]
I was successful configuring my instances of netdata according to your recommendations @thiagoftsm so thank you very much!

I was able to keep port `:19999` firewalled on all my machines and opened `20000` on the registry machine but blocked non-registry traffic by doing `bind to = *:20000=registry`. I think I can further tighten this by changing this from `bind to = *=registry` to `bind to = netdata2.mydomain.com=registry netdata3.mydomain.com=registry` but I haven't experimented with that yet. 

My nginx reverse proxy to `localhost:19999` is now configured with `auth_basic` and nodes can still report to the registry via `:20000`.

One thing I had to do was install an SSL cert on the registry so that there wasn't HTTPS mismatch for the nodes reporting from an https site (the node dashboard) to a non-https server (the registry). I installed a letsencrypt cert using a `$ letsencrypt --certonly` command. Now I have to figure out how to automate those renewals, but I will dig into that on my own as this isn't the forum for that.

One feature I'm missing, however, is the nice dashboard on netdata.cloud where all my nodes were listed side-by-side with their health status. Is there anyway to get a similar console dashboard when doing a local registry?

**Comment by thiagoftsm** [comment]
> Is there anyway to get a similar console dashboard when doing a local registry?

Netdata has custom dashboards, but I have to confess I never worked with it, so I cannot answer you. On the other hand we are releasing new cloud soon and I hope this also can help you. For while, I have to transfer your question for other coworker, because this is an area that I am not sure, so I prefer do not give answer, to avoid give you a wrong answer. @jacekkolasa , please, can you answer this question for us?  

Finally I am glad that I could help you, be free to ask you any doubt that you can have.

Best regards!

**Comment by jacekkolasa** [comment]
@TBNolan unfortunately the local registry takes care only of keeping track of visited nodes and redirecting after alarm. Have you tried the newly released cloud (1.22)? This makes problems with `auth_basic` irrelevant (metrics go through netdata.cloud, not directly from browser)
